app/Nova/Actions/Payments/RecordPaymentActions.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Nova\Actions\Payments;
+
+use App\Models\DuesTransaction;
+use App\Models\Payable;
+use App\Models\TravelAssignment;
+use Exception;
+use Illuminate\Support\Str;
+use Laravel\Nova\Actions\Action;
+use Laravel\Nova\Http\Requests\NovaRequest;
+
+/**
+ * Registers Nova actions for offline payment methods.
+ *
+ * @phan-suppress PhanCompatibleTraitConstant
+ * @phan-suppress PhanUnreferencedPrivateClassConstant
+ */
+trait RecordPaymentActions
+{
+    private const OFFLINE_PAYMENT_METHODS = [
+        RecordCashPayment::METHOD => RecordCashPayment::class,
+        RecordCheckPayment::METHOD => RecordCheckPayment::class,
+        ApplyWaiver::METHOD => ApplyWaiver::class,
+    ];
+
+    /**
+     * Get the actions available for the resource.
+     *
+     * @return array<\Laravel\Nova\Actions\Action>
+     */
+    public function actions(NovaRequest $request): array
+    {
+        $resourceType = $request->resource;
+        $resourceId = $request->resourceId ?? $request->resources;
+        $user = $request->user();
+
+        if ($resourceType === null || $resourceId === null || $user === null) {
+            return [];
+        }
+
+        if ($resourceType === \App\Nova\TravelAssignment::uriKey()) {
+            $payable = TravelAssignment::find($resourceId);
+        } elseif ($resourceType === \App\Nova\DuesTransaction::uriKey()) {
+            $payable = DuesTransaction::find($resourceId);
+
+            if ($payable === null || ! $payable->package->is_active) {
+                return [];
+            }
+        } else {
+            throw new Exception('Unexpected resource type '.$resourceType);
+        }
+
+        if ($payable === null || $payable->is_paid) {
+            return [];
+        }
+
+        $actions = [];
+
+        foreach (self::OFFLINE_PAYMENT_METHODS as $method => $class) {
+            if ($user->can('create-payments') && $user->can('create-payments-'.$method)) {
```

Comment:
Do we still need the separate `create-payments` permission?

Reply:
idk if there's much benefit if you can just grant the individual ones together

Reply:
The roles are already set up like that

Reply:
to be clear, I mean that as agreement that we don't need `create-payments`